File: scan-app/app/src/main/java/scan/ScanObservable.java
package scan;

import java.util.List;

import rx.Observable;
import rx.Subscriber;
import scan.listener.ScanListener;

/**
 * by y on 2017/2/15
 */

class ScanObservable implements Observable.OnSubscribe<List<AppModel>> {

    private int type = 0;
    private ScanListener listener = null;
    private boolean isNoSystem = false;

    ScanObservable(int type, ScanListener scanListener, boolean isNoSystem) {
        this.isNoSystem = isNoSystem;
        this.type = type;
        this.listener = scanListener;
    }

    @Override
    public void call(Subscriber<? super List<AppModel>> subscriber) {
        subscriber.onNext(QueryApp.getAppInfo(listener.getScanContext(), type, isNoSystem));
        subscriber.onCompleted();
    }
}

File: ScanApp/app/src/main/java/scan/ScanObservable.kt
package scan

import rx.Observable
import rx.Subscriber
import scan.listener.ScanListener

/**
 * by y on 2017/2/15
 */

class ScanObservable(private val type: Int, private val listener: ScanListener, private val isNoSystem: Boolean)
    : Observable.OnSubscribe<List<AppModel>> {

    override fun call(subscriber: Subscriber<in List<AppModel>>) {
        subscriber.onNext(QueryApp.getAppInfo(listener.scanContext, type, isNoSystem))
        subscriber.onCompleted()
    }
}
